packages/dds/tree/src/simple-tree/api/schemaFactoryAlpha.ts

```diff
@@ -307,6 +320,237 @@ export class SchemaFactoryAlpha<
 		>;
 	}
 
+	/**
+	 * Define a structurally typed {@link TreeNodeSchema} for a {@link (TreeRecordNode:interface)}.
+	 *
+	 * @param allowedTypes - The types that may appear in the record.
+	 *
+	 * @remarks
+	 * The identifier for this record is defined as a function of the provided types.
+	 * It is still scoped to this `SchemaFactory`, but multiple calls with the same arguments will return the same
+	 * schema object, providing somewhat structural typing.
+	 * This does not support recursive types.
+	 *
+	 * If using these structurally named records, other types in this schema builder should avoid names of the form `Record<${string}>`.
+	 *
+	 * @example
+	 * The returned schema should be used as a schema directly:
+	 * ```typescript
+	 * const MyRecord = factory.record(factory.number);
+	 * type MyRecord = NodeFromSchema<typeof Record>;
+	 * ```
+	 * Or inline:
+	 * ```typescript
+	 * factory.object("Foo", { myRecord: factory.record(factory.number) });
+	 * ```
+	 *
+	 * @privateRemarks
+	 * The name produced at the type-level here is not as specific as it could be; however, doing type-level sorting and escaping is a real mess.
+	 * There are cases where not having this full type provided will be less than ideal since TypeScript's structural types.
+	 * For example, attempts to narrow unions of structural records by name won't work.
+	 * Planned future changes to move to a class based schema system as well as factor function based node construction should mostly avoid these issues,
+	 * though there may still be some problematic cases even after that work is done.
```

Comment:
@CraigMacomber this is totally obsolete now, yes? Should we just remove this block (here and elsewhere)?

Reply:
I think this still applies to structurally named schema which are nor returned using class based typing to prevent invalid subclassing.